File: CS/CustomWizardExample/Wizard/Presenters/ChoosePersonsPage.cs
using System;
using System.Collections.Generic;
using DevExpress.DataAccess.Sql;
using DevExpress.DataAccess.Wizard;
using DevExpress.DataAccess.Wizard.Services;
using DevExpress.Entity.ProjectModel;
using DevExpress.XtraReports.Wizards;
using DevExpress.XtraReports.Wizards.Presenters;
using DevExpress.XtraReports.Wizards.Views;

namespace CustomWizardExample.Wizard.Presenters {
    class ChoosePersonsPage : ChooseReportTypePage<XtraReportModel> {
        public ChoosePersonsPage(IChooseReportTypePageView view, IConnectionStorageService connectionStorageService, 
            DataSourceTypes dataSourceTypes, IWizardRunnerContext context, ISolutionTypesProvider solutionTypesProvider)
            : base(view, connectionStorageService, dataSourceTypes, context, solutionTypesProvider) {
        }

        public override Type GetNextPageType() {
            return (View.ReportType == DevExpress.XtraReports.Wizards.ReportType.Standard)
                ? typeof(ChoosePersonsPage2<XtraReportModel>)
                : base.GetNextPageType();
        }
    }
}

File: VB/CustomWizardExample/Wizard/Presenters/ChoosePersonsPage.vb
Imports System
Imports System.Collections.Generic
Imports DevExpress.DataAccess.Sql
Imports DevExpress.DataAccess.Wizard
Imports DevExpress.DataAccess.Wizard.Services
Imports DevExpress.Entity.ProjectModel
Imports DevExpress.XtraReports.Wizards
Imports DevExpress.XtraReports.Wizards.Presenters
Imports DevExpress.XtraReports.Wizards.Views

Namespace CustomWizardExample.Wizard.Presenters
    Friend Class ChoosePersonsPage
        Inherits ChooseReportTypePage(Of XtraReportModel)

        Public Sub New(ByVal view As IChooseReportTypePageView, ByVal connectionStorageService As IConnectionStorageService, ByVal dataSourceTypes As DataSourceTypes, ByVal context As IWizardRunnerContext, ByVal solutionTypesProvider As ISolutionTypesProvider)
            MyBase.New(view, connectionStorageService, dataSourceTypes, context, solutionTypesProvider)
        End Sub

        Public Overrides Function GetNextPageType() As Type
            Return If(View.ReportType = DevExpress.XtraReports.Wizards.ReportType.Standard, GetType(ChoosePersonsPage2(Of XtraReportModel)), MyBase.GetNextPageType())
        End Function
    End Class
End Namespace
